### PR TITLE
genesis functions cleanup, testing

### PR DESCRIPTION
**Pull request by noot**
related to #313 

## Description

- fix ExportGenesis, add test
- update GenesisAccount code to custom type so we can json marshal into a hex string
- add InitGenesis test
______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/genesis.go [hunk 1]
```diff
@@ -29,7 +54,7 @@ type (
 	GenesisAccount struct {
 		Address ethcmn.Address   `json:"address"`
 		Balance *big.Int         `json:"balance"`
-		Code    []byte           `json:"code,omitempty"`
+		Code    Code             `json:"code,omitempty"`
```
Comment (fedekunze) [inline x/evm/types/genesis.go]: maybe we should use `ethcmn.Hex`? that would avoid us duplicating code

File: x/evm/types/statedb.go [hunk 2]
```diff
@@ -590,15 +590,15 @@ func (csdb *CommitStateDB) UpdateAccounts() {
 		currAcc := csdb.accountKeeper.GetAccount(csdb.ctx, sdk.AccAddress(addr.Bytes()))
 		emintAcc, ok := currAcc.(*emint.EthAccount)
 		if !ok {
-			return
+			continue
 		}
 
-		balance := csdb.bankKeeper.GetBalance(csdb.ctx, emintAcc.GetAddress(), emint.DenomDefault)
+		balance := csdb.bankKeeper.GetBalance(csdb.ctx, currAcc.GetAddress(), emint.DenomDefault)
```
Comment (fedekunze) [inline x/evm/types/statedb.go]: ni, this is technically the same one as before. So I'd undo the change

File: x/evm/types/utils.go [hunk 3]
```diff
@@ -165,3 +166,22 @@ func recoverEthSig(R, S, Vb *big.Int, sigHash ethcmn.Hash) (ethcmn.Address, erro
 
 	return addr, nil
 }
+
+// HexToBytes turns a 0x prefixed hex string into a byte slice
+func HexToBytes(in string) ([]byte, error) {
```
Comment (fedekunze) [inline x/evm/types/utils.go]: isn't there a util on ethereum for this function?

**Review by fedekunze — COMMENTED** [review]
Looks good in general 👍 I'd use `ethcmn.Hex` for the `Code` genesis field to avoid duplication.

File: x/evm/types/genesis.go (hunk 1 above)
Comment (noot) [reply x/evm/types/genesis.go]: ah good move, will update

File: x/evm/types/utils.go (hunk 3 above)
Comment (noot) [reply x/evm/types/utils.go]: yeah probably, I won't need it once I switch to ethcmn.Hex

File: x/evm/types/msg.go [hunk 4]
```diff
@@ -63,6 +63,11 @@ func NewMsgEthermint(
 	}
 }
 
+func (msg MsgEthermint) String() string {
```
Comment (fedekunze) [inline x/evm/types/msg.go:66]: we should use YAML Marshal for `String()` in the future. I'll create an issue.

**Review by fedekunze — APPROVED** [review]
utACK
